src/components/TaskStatusIcon/index.tsx

```diff
@@ -0,0 +1,50 @@
+import { uiColors } from "@leafygreen-ui/palette";
+import Icon, { Size } from "components/Icon";
+import { TaskStatus } from "types/task";
+import { errorReporting } from "utils";
+
+const { reportError } = errorReporting;
+const { green, red, yellow, gray } = uiColors;
+
+interface TaskStatusIconProps {
+  status: TaskStatus;
+  size?: Size | number;
+}
+
+export const TaskStatusIcon: React.FC<TaskStatusIconProps> = ({
+  status,
+  size = Size.Default,
+}) => {
+  switch (status) {
+    case TaskStatus.Succeeded:
+      return <Icon glyph="Checkmark" fill={green.base} size={size} />;
+    case TaskStatus.Failed:
+      return <Icon glyph="X" fill={red.base} size={size} />;
+    case TaskStatus.KnownIssue:
+      return <Icon glyph="KnownFailureIcon" fill={red.base} size={size} />;
+    case TaskStatus.Dispatched:
+    case TaskStatus.Started:
+      return <Icon glyph="Refresh" fill={yellow.dark2} size={size} />;
+    case TaskStatus.SetupFailed:
+      return <Icon glyph="SetupFailure" fill={red.base} size={size} />;
+    case TaskStatus.SystemUnresponsive:
+    case TaskStatus.SystemTimedOut:
+    case TaskStatus.SystemFailed:
+      return <Icon glyph="SystemFailure" fill="#36367F" size={size} />;
```

Comment:
[nit] Could we assign this to a variable so it's a little more clear this is purple?